### PR TITLE
feat: Add support for random password generation and Terragrunt wrapper

### PR DESCRIPTION
**Pull request by bryantbiggs**
## Description
- Add support for random password generation and Terragrunt wrapper

## Motivation and Context
- Rather than passing in a password string, allow users to generate a strong password that is stored directly in the secret
- Support Terragrunt wrappers approach

## Breaking Changes
- No

## How Has This Been Tested?
- [x] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [x] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by bryantbiggs** [comment]
cc @antonbabenko if you get time

File: variables.tf [hunk 1]
```diff
@@ -118,6 +118,24 @@ variable "version_stages" {
...
+variable "random_password_length" {
+  description = "The length of the generated random password"
+  type        = number
+  default     = 32
+}
+
+variable "random_password_override_special" {
+  description = "Supply your own list of special characters to use for string generation. This overrides the default character list in the special argument"
+  type        = string
+  default     = "!#$%&*()-_=+[]{}<>:?"
```
Comment (antonbabenko) [inline variables.tf:136]: Do you know where these chars came from? There are no `/.,\~`, for example. Is there some kind of document containing such a set?

**Review by antonbabenko — APPROVED** [review]
LGTM otherwise.

File: variables.tf (hunk 1 above)
Comment (bryantbiggs) [reply variables.tf]: whoops - looks like I copied the example value and not the actual default value. Those are coming from here https://registry.terraform.io/providers/hashicorp/random/latest/docs/resources/password#special

I believe the others you listed tend to cause issues in (I know some of those are not supported in RDS secrets depending on the DB engine)

Comment (bryantbiggs) [reply variables.tf]: also it looks like I don't have permission on this repo yet if you wouldn't mind adding/merging when you get a chance

Comment (antonbabenko) [reply variables.tf]: I've just fixed permissions and merged.

**Comment by antonbabenko** [comment]
This PR is included in version 1.1.0 :tada:

**Pull request merged**